### DISCUSSION TITLE
Shared playlist quick save/auto save to file

### DISCUSSION THREAD

**Discussion by enchained** [Syncplay/syncplay#447] [Ideas]
I downloaded what I think is the latest dev build, cause it has the feature of loading/saving shared playlists, that soon will surely be a very welcome in our remote household. But I immediately thought about how could it possibly be improved. 

Right now to save the playlist changes I have to use the save window dialog and overwrite the file. However, if the file was previously saved/loaded, an auto-save (configurable option), or at least a quick save (new context menu option that overwrites the current file without prompting, something acting like Ctrl+S and with key binding) would surely be a nice addition. 

Also, on startup, when the playlist is empty, it would be very useful to auto-load the playlist, if it was already saved before while in that room.

As the last thought, specifying a default playlist folder in the settings could improve the existing saving experience, since currently after the restart it looks like the saving dialog starts in the folder of the media file that was loaded.

**Comment by Et0h** [comment]
Some changes would be relatively straightforward, like allowing users to specify a default playlist folder. Others require more complicated choices.

To know how auto-saving and auto-loading would best work I would really need to get more comments on how users would want such a feature to work in terms of the precise behaviour. For example, is this what people want:
* You can turn on/off shared playlists from settings. This is blanket on/off.
* Playlists stored in specified playlist folder, which is then default for save/load. By default could be a /Syncplay Playlists/ sub-folder in your documents folder or perhaps in your %APPDATA% folder.
* Playlists in autoload/autosave feature stored with filename such as 'syncplay.pl 8998 myroom1.m3u8, specifying the server name, port and room name.
* If you join a room (including by joining a server and having that room as default) it will use the shared playlist which exists, unless it is empty in which case it will check for the .m3u8 file. If the .m3u8 file exists it will load the playlist automatically.
* If you join a room with a playlist an existing playlist it will auto-save that playlist when you join and whenever it changes, even if that means overwriting your previously saved playlist.

Now for me the fact that it would automatically overwrite your old playlist could be seen as somewhat problematic. There could be some sort of backup system, but in that case when would the backup be saved? Or would we have to make it require the user to check a checkbox every time to start auto-saving the playlist as a workaround, in which case how often would people accidentally forget? Or should it just be a button people would have to manually press to save to the right folder and filename to avoid this issue? Then it wouldn't be auto-save, but it would at least be relatively predictable as a behaviour.

One possibility then is to have a 'load' and 'save' playlist button which load and save to the standardised filename in the specified/default folder, but if people want to they can also set it to auto-load the playlist if the playlist is empty.

Also, in terms of file names do people prefer .m3u8 or .txt?

**Comment by Latelistener** [comment]
I think it would be possible to deal with the unexpected playlist overwriting by auto-saving it just once on exit (and auto-loading it if the playlist is empty). There are various ways I can think saving on exit could be implemented. Either by marking a checkbox, let's say, somewhere near the "Enabled shared playlists" box or somewhere in the options (preferably near the field where you will specify the default folder for playlists). Alternatively, Syncplay could also ask you whether to save playlist or not on exit, but it's somewhat more intrusive and should probably also be optional.

Quick save should be either in the drop-down menu when you click on the playlist field or as a small button somewhere near the playlist itself.

As for the default playlist folder initial value, I would prefer Syncplay to save it in its own folder (or a "playlists" subfolder inside). At least if you're using the portable version.

As for the extension, I would prefer .m3u8 which is a simple and elegant way to tell you that it's a playlist file. And you can still view and edit those with any text editor.

**Comment by MarisKori** [comment]
I'd like to have a checkbox (which is called "auto-save") under/near shared playlist in Syncplay window.
If it's checked, Syncplay saves the shared list 10 seconds after the last change and before quit.
Of course, this checkbox should also be part of the settings. So changing the checkbox also changes the settings (and saves it).

**Comment by Et0h** [comment]
I'm currently considering having auto save/load playlist in client-side persistent rooms linked to the user who create the room. See #434 for the relevant discussion.
